File: fabric/common/src/main/java/com/noxcrew/noxesium/fabric/network/clientbound/ClientboundResetExtraEntityDataPacket.java
package com.noxcrew.noxesium.fabric.network.clientbound;

import com.noxcrew.noxesium.api.fabric.network.NoxesiumPacket;
import com.noxcrew.noxesium.api.fabric.network.NoxesiumPayloadType;
import com.noxcrew.noxesium.fabric.network.CommonPackets;
import it.unimi.dsi.fastutil.ints.IntList;
import net.minecraft.network.RegistryFriendlyByteBuf;
import net.minecraft.network.codec.StreamCodec;
import net.minecraft.network.protocol.common.custom.CustomPacketPayload;

/**
 * Resets the stored value for extra data on an entity.
 */
public record ClientboundResetExtraEntityDataPacket(int entityId, IntList indices) implements NoxesiumPacket {
    public static final StreamCodec<RegistryFriendlyByteBuf, ClientboundResetExtraEntityDataPacket> STREAM_CODEC =
            CustomPacketPayload.codec(
                    ClientboundResetExtraEntityDataPacket::write, ClientboundResetExtraEntityDataPacket::new);

    private ClientboundResetExtraEntityDataPacket(RegistryFriendlyByteBuf buf) {
        this(buf.readVarInt(), buf.readIntIdList());
    }

    private void write(RegistryFriendlyByteBuf buf) {
        buf.writeVarInt(entityId);
        buf.writeIntIdList(indices);
    }

    @Override
    public NoxesiumPayloadType<?> noxesiumType() {
        return CommonPackets.INSTANCE.CLIENT_RESET_EXTRA_ENTITY_DATA;
    }
}

File: paper/src/main/kotlin/com/noxcrew/noxesium/paper/api/network/clientbound/ClientboundResetExtraEntityDataPacket.kt
package com.noxcrew.noxesium.paper.api.network.clientbound

import com.noxcrew.noxesium.paper.api.network.NoxesiumPacket
import com.noxcrew.noxesium.paper.api.network.NoxesiumPackets
import it.unimi.dsi.fastutil.ints.IntList

/**
 * Resets the stored value for extra data on an entity.
 */
public class ClientboundResetExtraEntityDataPacket(
    public val entityId: Int,
    public val indices: IntList,
) : NoxesiumPacket(NoxesiumPackets.CLIENT_RESET_EXTRA_ENTITY_DATA)
